onnxruntime/core/graph/function.cc

```diff
@@ -177,22 +194,12 @@ static std::unique_ptr<ONNX_NAMESPACE::OpSchema> CreateSchema(const Graph& graph
   return op_schema;
 }
 
-// Creates domain to version map for onnx function by merging graph level opset imports with opset imports from
-// funtion proto
-static std::unordered_map<std::string, int> CreateOpsetImportsForFunction(const ONNX_NAMESPACE::FunctionProto& func_proto,
-                                                                          const std::unordered_map<std::string, int>& graph_opset_imports) {
-  // function inherits all graph level opset imports
-  std::unordered_map<std::string, int> function_opset_imports{graph_opset_imports};
-  // merge with opset imports in function proto
+// Creates domain to version map for onnx function
+static std::unordered_map<std::string, int> GetFunctionOpsetImports(const ONNX_NAMESPACE::FunctionProto& func_proto, const std::unordered_map<std::string, int>& graph_imports) {
+  std::unordered_map<std::string, int> function_opset_imports{graph_imports};
   for (const auto& opset_import : func_proto.opset_import()) {
-    auto opset_version = static_cast<int>(opset_import.version());
-    auto result = function_opset_imports.insert({opset_import.domain(), opset_version});
-    ORT_ENFORCE((result.first->second == opset_version),
-                "ONNX model does not support multiple opset versions for a domain. Model imports opset version ",
-                result.first->second, " for domain ", result.first->first, " and function is trying to import opset version ",
-                opset_version, " for the same domain");
+    function_opset_imports.insert({opset_import.domain(), static_cast<int>(opset_import.version())});
```

Comment:
May be worth adding a line of comment (and pointer to the other compatibility check): is it a precondition that the graph-imports and function-imports don't conflict at this point?

Reply:
Nope it is not.... We simply use the opset imports provided as part of function proto... If ORT does not have a specialized kernel for this function op then during function inlining any discrepancy will be caught.